internal/controller/controllers/clusterdeployments_controller.go

```diff
@@ -702,6 +702,10 @@ func getPlatform(openshiftPlatformType hiveext.PlatformType) *models.Platform {
 		return &models.Platform{
 			Type: common.PlatformTypePtr(models.PlatformTypeBaremetal),
 		}
+	case hiveext.ExternalPlatformType:
+		return &models.Platform{
+			Type: common.PlatformTypePtr(models.PlatformTypeOci),
```

Comment:
In the long term I think we'll probably want to add something like an `externalPlatformName` field so that we can conditionally return a different platform here. That will require us to add a vanilla 'external' platform type to core assisted. (The difference between `oci` and `external` would be what validations are run.)

The question for me is what to do in the short term - there are several options:

1. Add "External" to the k8s API as proposed here. A downside of this is that on this line we default to PlatformTypeOci, but once there is an `externalPlatformName` field then oci will not be the default. So we are creating a future backward-compat problem.
2. Add "OCI" to the k8s API instead. A downside of this is that ZTP does not and probably never will deploy to OCI, so it's a little weird for this to appear as a first-class option in the k8s API.
3. Go ahead and add the vanilla 'external' platform and the `externalPlatformName` field now. A downside of this is that it will take time that we don't really have.

Is there another option I have missed?

Reply:
@adriengentil said he is going to explore the 3rd option, which is best technically, provided it doesn't take a lot of time.

Reply:
Playing a bit here: https://github.com/openshift/assisted-service/pull/5443